crates/net/network/src/transactions/fetcher.rs

```diff
@@ -160,7 +160,7 @@ impl TransactionFetcher {
                 }
             }
         };
-        let hash = hashes_to_request.first()?;
+        let hash = hashes_to_request.iter().next()?;
```

Comment:
This won't be deterministic anymore, don't know if that matters here but it's something to note.

Reply:
at this point `hashes_to_request` will only contain one hash